src/QMCWaveFunctions/BsplineFactory/EinsplineSetBuilderCommon.cpp

```diff
@@ -174,9 +174,12 @@ void EinsplineSetBuilder::BroadcastOrbitalInfo()
   if (primcell_kpoints.size() != NumTwists)
     primcell_kpoints.resize(NumTwists);
   bbuffer.add(&primcell_kpoints[0][0], &primcell_kpoints[0][0] + OHMMS_DIM * NumTwists);
-  bibuffer.add(&(TargetPtcl.DensityReducedGvecs[0][0]),
-               &(TargetPtcl.DensityReducedGvecs[0][0]) + numDensityGvecs * OHMMS_DIM);
-  bbuffer.add(&(TargetPtcl.Density_G[0]), &(TargetPtcl.Density_G[0]) + numDensityGvecs);
+  if (numDensityGvecs > 0)
+  {
+    bibuffer.add(&(TargetPtcl.DensityReducedGvecs[0][0]),
```

Comment:
I'm going to ignore the issue of the details of the double vector here. 
If the problem is an assert when `vector.operator[](0) when vector.empty()==true` happens how does this help? You just don't add a 0 length segment to bibuffer and bbuffer.  If numDensityGvecs <=0 has the side effect that TargtetPtcl.DensityReducedGvecs is empty why not be clear and just test that?

Reply:
It's clear enough for me now.